spmd/tensor/api.py

```diff
@@ -1,30 +1,41 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
 import copy
 import math
+
 import torch
 from torch.utils._pytree import tree_map, tree_flatten
 from typing import Dict, Callable, Optional, Sequence
 from spmd.tensor.device_mesh import get_global_device_mesh, DeviceMesh
-from spmd.tensor.placement_types import Placement, Shard, Replicate, _Partial
+from spmd.tensor.placement_types import (
+    Placement,
+    Shard,
+    Replicate,
+    _Partial,
+    PlacementSpec,
+)
 from spmd.tensor.redistribute import Redistribute
 
-
-"""
-If set to true, __DEBUG_STRICT will fail when an op doesn't have a sharding rule registered.
-"""
-_DEBUG_STRICT = False
+from spmd.tensor.utils import (
+    unwrap_local_tensor,
+    unwrap_spec,
+)
+from spmd.tensor.dispatch import OpInfo, dispatch_operator
 
 
 class DTensor(torch.Tensor):  # pyre-ignore[13]: pyre is bad at __new__
     _local_tensor: torch.Tensor
-    _device_mesh: DeviceMesh
-    _placements: Sequence[Placement]
-    __slots__ = ["_local_tensor", "_device_mesh", "_placements"]
+    _placement_spec: PlacementSpec
+    __slots__ = ["_local_tensor", "_placement_spec"]
+
+    # class attribute that handles operator placements propagation
+    # rules, keyed by aten op name, value is propagation func
+    _op_to_rules: Dict[str, Callable] = {}
 
-    # class attribute that handles ops, all handled
-    # ops should appear in this table
+    # class attribute that handles custom registered ops, all handled
+    # custom ops should appear in this table, and overriding the default
+    # operators
     # pyre-fixme[24]: Generic type `Callable` expects 2 type parameters.
```

Comment:
Could you actually fix this -- it would help document the callable's semantic via its signatrue.